src/Symfony/Component/JsonEncoder/Decode/LazyInstantiator.php

```diff
@@ -44,34 +44,37 @@ final class LazyInstantiator
     private static array $lazyClassesLoaded = [];
 
     public function __construct(
-        private string $lazyGhostsDir,
+        private ?string $lazyGhostsDir = null,
     ) {
-        $this->fs = new Filesystem();
+        if (null === $this->lazyGhostsDir && \PHP_VERSION_ID < 80400) {
+            throw new InvalidArgumentException('$lazyGhostsDir must be provided for PHP versions under 8.4.');
```

Comment:
Generally, this wording is used:

```suggestion
            throw new InvalidArgumentException('The "lazyGhostsDir" argument cannot be null on PHP < 8.4.');
```

"must" should never be used in user-oriented messages, and "</>/<=/>=" are usually used also across the codebase I think when talking about PHP and packages versions.

Reply:
we have plenty of "must" in user messages across the codebase so this would need to be discussed elsewhere, but suggestion looks good

Reply:
Ah the leading `$` should be kept

Reply:
Indeed, I know `symfony-docs` bans it if I remember correctly but nothing is clearly written in the coding standards. Here's a better place to discuss about it: https://github.com/symfony/symfony-docs/pull/20468